### PR TITLE
Correctly handle MD health if no replicas are configured

### LINKED ISSUE #5834: MachineDeployments with zero replicas show as "Updating"
### What happened

_This was reported by a customer originally._

When a MachineDeployment is scaled to zero replicas and the node is gone already, the MachineDeployment still shows as "Updating". This is incorrect since nothing is happening with the MachineDeployment and it has reached its expected state:

Screenshot 2023-03-14 at 14 32 31

It would be great to backport a fix to this to v2.22 and v2.21 since this is a customer-reported issue.

### Expected behavior

MachineDeployments with zero replicas should show as "Ready" (or whatever the green bubble stands for).

### How to reproduce

Create MachineDeployment with zero replicas in a KKP user cluster. Observe the status bubble to be a spinner, saying "Updating".

### Environment
- UI Version: v2.22.0 & v2.21.6
- API Version: v2.22.0 & v2.21.6
- Domain:
- Others:

### Current workaround

No known workaorund.

### Affected user persona

Cluster owners

### Metric to be improved

UI correctness

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:

When a MachineDeployment has zero replicas, the returned status does not contain any information about replica health (since the value is zero and is omitted by the backend's struct definition) and thus, the dashboard shows it as "Updating".

Since we cannot change the backend code (it uses the cluster-api definition of MachineDeployment status), I tried fixing this in the frontend; I couldn't think of a better way to handle this than including a "special case" in the if condition. If there's a way to simplify it that I've missed, I'm very open to suggestions.

**Which issue(s) this PR fixes**:

Fixes #5834

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Show correct health information for Machine Deployments with no replicas
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/app/shared/utils/health-status.ts [hunk 1]
```diff
@@ -85,9 +85,9 @@ export function getMachineDeploymentHealthStatus(md: MachineDeployment): HealthS
     return new HealthStatus(StatusMassage.Deleting, StatusIcon.Error);
   } else if (
     md.status &&
-    md.status.availableReplicas === md.spec.replicas &&
-    md.status.availableReplicas === md.status.updatedReplicas
-  ) {
+    ((md.status.availableReplicas === md.spec.replicas &&
+    md.status.availableReplicas === md.status.updatedReplicas) ||
+    (md.spec.replicas === 0 && !md.status.availableReplicas && !md.status.updatedReplicas))) {
```
Comment (ahmedwaleedmalik) [inline modules/web/src/app/shared/utils/health-status.ts:90]:
```suggestion
    md.status &&
    (md.status.availableReplicas || 0) === md.spec.replicas &&
    md.status.availableReplicas === md.status.updatedReplicas
```

Comment (embik) [reply modules/web/src/app/shared/utils/health-status.ts:90]: Much better, thanks!

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**